code/modules/organs/internal/_internal.dm

```diff
@@ -3,96 +3,80 @@
+/obj/item/organ/internal/Initialize(mapload, material_key, datum/dna/given_dna)
+	if(!alive_icon)
+		alive_icon = initial(icon_state)
 	if(max_damage)
 		min_bruised_damage = FLOOR(max_damage / 4)
 	. = ..()
-	if(. != INITIALIZE_HINT_QDEL && owner)
-		owner.internal_organs |= src
+
+//Returns FALSE if we should abort init
+/obj/item/organ/internal/try_autoinstall()
+	if(owner)
 		var/obj/item/organ/external/E = owner.get_organ(parent_organ)
 		if(!E)
 			PRINT_STACK_TRACE("[src] spawned in [owner] without a parent organ: [parent_organ].")
-			return INITIALIZE_HINT_QDEL
-		LAZYDISTINCTADD(E.internal_organs, src)
-		E.cavity_max_w_class = max(E.cavity_max_w_class, w_class)
-		E.update_internal_organs_cost()
+			return FALSE
+		owner.add_organ(src, E)
+	return TRUE
 
-/obj/item/organ/internal/Destroy()
-	if(owner)
-		owner.internal_organs -= src
-		owner.internal_organs_by_name -= organ_tag
-		var/obj/item/organ/external/E = owner.organs_by_name[parent_organ]
-		if(istype(E))
-			LAZYREMOVE(E.internal_organs, src)
-	return ..()
-
-/obj/item/organ/internal/set_dna(var/datum/dna/new_dna)
-	..()
-	if(species && species.organs_icon)
+/obj/item/organ/internal/set_species(species_name)
+	. = ..()
+	if(species.organs_icon)
 		icon = species.organs_icon
 
 //disconnected the organ from it's owner but does not remove it, instead it becomes an implant that can be removed with implant surgery
 //TODO move this to organ/internal once the FPB port comes through
 /obj/item/organ/proc/cut_away(var/mob/living/user)
 	var/obj/item/organ/external/parent = owner.get_organ(parent_organ)
 	if(istype(parent)) //TODO ensure that we don't have to check this.
-		removed(user, 0)
+		uninstall(detach = TRUE)
 		LAZYADD(parent.implants, src)
 
-/obj/item/organ/internal/removed(var/mob/living/user, var/drop_organ=1, var/detach=1)
-	if(owner)
-		owner.internal_organs -= src
-		owner.internal_organs_by_name -= organ_tag
-		var/obj/item/organ/external/E = owner.get_organ(parent_organ)
-		E.update_internal_organs_cost()
-
-		if(detach)
-			var/obj/item/organ/external/affected = owner.get_organ(parent_organ)
-			if(affected)
-				LAZYREMOVE(affected.internal_organs, src)
-				status |= ORGAN_CUT_AWAY
-	..()
-
-/obj/item/organ/internal/replaced(var/mob/living/carbon/human/target, var/obj/item/organ/external/affected)
-
-	if(!istype(target))
-		return 0
-
+/obj/item/organ/internal/install(mob/living/carbon/human/target, obj/item/organ/external/affected, in_place)
 	if(status & ORGAN_CUT_AWAY)
-		return 0 //organs don't work very well in the body when they aren't properly attached
+		return
 
+	//#FIXME: This feels like a hack
+		//# Also it probably should be of the species of the parent limb tbh..
 	// robotic organs emulate behavior of the equivalent flesh organ of the species
-	if(BP_IS_PROSTHETIC(src) || !species)
-		species = target.species
+	// if(BP_IS_PROSTHETIC(src) || !species)
+	// 	set_species(target.species)
```

Comment:
What's the status on this?

Reply:
Oh right! I left that commented because I was trying to see if it was actually needed. Because with the current setup, organs are created with dna info, and get the specie through it already. 
So, changing this post-init could cause unforeseen problems. XD

I'm just a bit unsure on the use cases here. Because, this bit of code would mean if we have prosthetic alien organs they could be slapped into any races without the downside that comes with the real thing.. And organs transplanting and prosthesis are already very buggy as it is. 
Simplest way imo is to just create them for a target specie, so validation can be done at whatever created it.

Honestly, I feel like overall the way prosthetics works might need some rethinking? I'd be down to make some changes if it comes to that.

Reply:
Decided to remove that code, since it was a hacky and counter productive way to guarantee something that we're already guaranteeing before we get to this point.